product.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+"""
+    product.py
+
+    :copyright: (c) 2015 by Openlabs Technologies & Consulting (P) Limited
+    :license: see LICENSE for more details.
+"""
+from trytond.pool import PoolMeta
+from trytond.model import ModelView, fields, ModelSQL
+
+__metaclass__ = PoolMeta
+__all__ = ['ProductSaleChannelListing', 'Product']
+
+
+class Product:
+    "Product"
+    __name__ = "product.product"
+
+    channel_listings = fields.One2Many(
+        'product.sale.channel_listing', 'product', 'Channel Listings',
+    )
+
+
+class ProductSaleChannelListing(ModelSQL, ModelView):
+    '''Product - Sale Channel
+    This model keeps a record of a product's association with Sale Channels.
+    A product can be listen on multiple marketplaces
+    '''
+    __name__ = 'product.sale.channel_listing'
+
+    # TODO: Only show channels where this ability is there. For example
+    # showing a manual channel is pretty much useless
+    channel = fields.Many2One(
+        'sale.channel', 'Sale Channel',
+        domain=[('source', '!=', 'manual')],
+        required=True, select=True,
+        ondelete='RESTRICT'
+    )
+    product = fields.Many2One(
+        'product.product', 'Product', required=True, select=True,
+        ondelete='CASCADE'
+    )
```

Comment:
add a field `product_identifier` that is a Char field. It should store the reference of the product as understood by the external system.